### ISSUE TITLE
Implement ScriptScoreQuery

### ISSUE THREAD

**Issue by sam-briggs-depop** [sksamuel/elastic4s#2941]
# TLDR;

I'd like to use the ElasticSearch [script score query ](https://www.elastic.co/guide/en/elasticsearch/reference/8.10/query-dsl-script-score-query.html). I don't believe it's implemented in `elastic4s`. Would it be acceptable to create a PR that implements it?

# Details

The ElasticSearch Search DSL supports a script score query since v7.0. See docs for [v8.10 (latest)](https://www.elastic.co/guide/en/elasticsearch/reference/8.10/query-dsl-script-score-query.html) and [v7.0](https://www.elastic.co/guide/en/elasticsearch/reference/7.0/query-dsl-script-score-query.html).

The example HTTP request from the docs looks like:

```
GET /_search
{
  "query": {
    "script_score": {
      "query": {
        "match": { "message": "elasticsearch" }
      },
      "script": {
        "source": "doc['my-int'].value / 10 "
      }
    }
  }
}
```

As far as I can see, elastic4s doesn't support constructing such a query. There _does_ exist the [`ScriptScoreQuery` class](https://github.com/sksamuel/elastic4s/blob/master/elastic4s-domain/src/main/scala/com/sksamuel/elastic4s/requests/searches/queries/ScriptScoreQuery.scala) but the writing of this class is not implemented in [`QueryBuilderFn`](https://github.com/sksamuel/elastic4s/blob/bff9d655b04f52cdf7f2c08c72dcee2e51157620/elastic4s-handlers/src/main/scala/com/sksamuel/elastic4s/handlers/searches/queries/QueryBuilderFn.scala#L59).

I found this stackoverflow from a user in 2020 searching for the same functionality: https://stackoverflow.com/questions/65238974/scriptscorequery-implementation-in-elastic4s

**Comment by sam-briggs-depop** [comment]
p.s. thanks for such a wonderful high level client. I really enjoy developing using this tool <3

**Comment by sksamuel** [comment]
Once your PR is ready I will merge and release.
